### PR TITLE
avoid ax-8 in nfceqdf

### LINKED ISSUE #4156: Please commit for me
Hello,

I still have no replacement for my broken notebook, so maybe someone can help
out here, and provide a commit to github.  Theorem nfceqdf is not necessarily
dependent on ax-8, as can be seen in the following proof.  This has a limited
cascading effect, as nfcxfrdf, nfded, nfded2 in Norm's mathbox lose this
dependency as well.

Theorem nfceqdf
Description: An equality theorem for effectively not free. (Contributed by Mario Carneiro, 14-Oct-2016.)
             Avoid ax-8 .  (Revised by Wolf Lammen, 23-08-2024)

Hypotheses
nfceqdf.1  ⊢ Ⅎ𝑥𝜑
nfceqdf.2  ⊢ (𝜑 → 𝐴 = 𝐵)

distinct variables: $d(𝑦,𝑥)  $d(𝑦,𝜑)  $d(𝑦,𝐴)  $d(𝑦,𝐵)

Assertion
nfceqdf  ⊢ (𝜑 → (Ⅎ𝑥𝐴 ↔ Ⅎ𝑥𝐵))

1                 nfceqdf.1  ⊢ Ⅎ𝑥𝜑
2                 nfceqdf.2  ⊢ (𝜑 → 𝐴 = 𝐵)
3                 dfcleq        ⊢ (𝐴 = 𝐵 ↔ ∀𝑦(𝑦 ∈ 𝐴 ↔ 𝑦 ∈ 𝐵))
4   2,3         sylib          ⊢ (𝜑 → ∀𝑦(𝑦 ∈ 𝐴 ↔ 𝑦 ∈ 𝐵))
5   4            19.21bi     ⊢ (𝜑 → (𝑦 ∈ 𝐴 ↔ 𝑦 ∈ 𝐵))
6   1,5         nfbidf       ⊢ (𝜑 → (Ⅎ𝑥 𝑦 ∈ 𝐴 ↔ Ⅎ𝑥 𝑦 ∈ 𝐵))
7   6            albidv       ⊢ (𝜑 → (∀𝑦Ⅎ𝑥 𝑦 ∈ 𝐴 ↔ ∀𝑦Ⅎ𝑥 𝑦 ∈ 𝐵))
8                 df-nfc       ⊢ (Ⅎ𝑥𝐴 ↔ ∀𝑦Ⅎ𝑥 𝑦 ∈ 𝐴)
9                 df-nfc       ⊢ (Ⅎ𝑥𝐵 ↔ ∀𝑦Ⅎ𝑥 𝑦 ∈ 𝐵)
10  7,8,9    3bitr4g    ⊢ (𝜑 → (Ⅎ𝑥𝐴 ↔ Ⅎ𝑥𝐵))

### PR DESCRIPTION
**Pull request by icecream17**
Fixes #4156 

@benjub eleq2w2 is stronger than [bj-ax9](https://us.metamath.org/mpeuni/bj-ax9.html)

### CODE REVIEW

**Comment by wlammen** [comment]
Modifications approved

File: set.mm [hunk 1]
```diff
@@ -26991,8 +27004,18 @@ generally appear in a single form (either definitional, but more often
     nfceqdf.1 $e |- F/ x ph $.
     nfceqdf.2 $e |- ( ph -> A = B ) $.
     $( An equality theorem for effectively not free.  (Contributed by Mario
-       Carneiro, 14-Oct-2016.) $)
+       Carneiro, 14-Oct-2016.)  Avoid ~ ax-8 .  (Revised by WL and SN,
```
Comment (benjub) [inline set.mm:27007]:
```suggestion
       Carneiro, 14-Oct-2016.)  Avoid ~ ax-8 and ~ df-clel .  (Revised by WL and SN,
```

File: set.mm [hunk 2]
```diff
@@ -25458,6 +25458,19 @@ the definition of class equality ( ~ df-cleq ).  Its forward implication
...
+    $d A x $.  $d B x $.
+    $( A weaker version of ~ eleq2 that uses ~ ax-12 to avoid ~ ax-8 and
+       ~ df-clel .  Compare ~ eleq2w , whose setvars appear where the class
+       variables are in this theorem, and vice versa.  (Contributed by WL and
+       SN, 23-Aug-2024.)  (Proof modification is discouraged.)
+       (New usage is discouraged.) $)
+    eleq2w2 $p |- ( A = B -> ( x e. A <-> x e. B ) ) $=
+      ( wceq cv wcel wb wal dfcleq biimpi 19.21bi ) BCDZAEZBFMCFGZALNAHABCIJK
+      $.
+    $( $j usage 'eleq2w2' avoids 'ax-8' 'df-clel'; $)
```
Comment (benjub) [inline set.mm:25471]:
```suggestion
    $( $j usage 'eleq2w2' avoids 'ax-8' 'ax-10' 'ax-11' 'ax-13' 'df-clab' 'df-clel'; $)
```

File: set.mm [hunk 3]
```diff
@@ -25458,6 +25458,19 @@ the definition of class equality ( ~ df-cleq ).  Its forward implication
        'df-clel'; $)
   $}
 
+  ${
+    $d A x $.  $d B x $.
+    $( A weaker version of ~ eleq2 that uses ~ ax-12 to avoid ~ ax-8 and
+       ~ df-clel .  Compare ~ eleq2w , whose setvars appear where the class
+       variables are in this theorem, and vice versa.  (Contributed by WL and
+       SN, 23-Aug-2024.)  (Proof modification is discouraged.)
+       (New usage is discouraged.) $)
```
Comment (benjub) [inline set.mm:25467]: Why discouragement tags ?

**Comment by benjub** [comment]
> Fixes #4156
> 
> @benjub eleq2w2 is stronger than [bj-ax9](https://us.metamath.org/mpeuni/bj-ax9.html)

Correct. And the proofs are basically the same (with `eleq2w2`'s shorter since there is no need to weaken the biconditional to an implication). Actually, `bj-ax9` was stated like that to be identical to `ax-9`.

So, I think `eleq2w2` should replace `bj-ax9`, especially in the comment of `ax9ALT`.  At the time (2019), it was deemed that `bj-ax9` could stay in the mathbox (although I wanted to move it to Main).  So now, if `eleq2w2` replaces it, we kind of lose my work on that earlier version.

Is it possible to credit `eleq2w2` as:
```
   (Contributed by BJ, 24-Jun-2019.)  Strengthen from setvar variables to
   class variables. (Revised by WL and SN, 23-Aug-2024.) $)
```
?

This sounds reasonable to me, but we need approval of @icecream17 and @wlammen.

**Comment by wlammen** [comment]
@benjub fine with me

**Review by avekens — APPROVED** [review]

**Review by benjub — APPROVED** [review]

**Review by benjub — APPROVED** [review]

**Pull request merged**